File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dict = {}
    for p in participant:
        dict[p] = dict.get(p, 0) + 1
        
    for c in completion:
        dict[c] -= 1
    
    for key, value in dict.items():
        if value == 1:
            answer += key
            break
    
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> sh = new HashMap<>();
        for (String key : participant) {
            sh.put(key, sh.getOrDefault(key, 0) + 1);
        }
        for (String key : completion) {
            sh.put(key, sh.get(key) - 1);
            if (sh.get(key) == 0) sh.remove(key);
        }
        for (String key : sh.keySet()) {
            answer = key;
        }
        return answer;
    }
}